_data-prepper/pipelines/configuration/sinks/opensearch.md

```diff
@@ -29,16 +67,219 @@ index | Conditionally | String | Name of the export index. Applicable and requir
+
+For instructions to map users to roles, see [Map users to roles]({{site.url}}{{site.baseurl}}/security/access-control/users-roles/#map-users-to-roles).
+
+## Amazon OpenSearch Service domain security
+
+The OpenSearch sink plugin can send data to an [Amazon OpenSearch Service](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/what-is.html) domain, which uses Identity and Access Management (IAM) for security. The plugin uses the default credential chain. Run `aws configure` using the [AWS CLI](https://aws.amazon.com/cli/) to set your credentials.
+
+Make sure the credentials that you configure have the required IAM permissions. The following domain access policy demonstrates the minimum required permissions for the sink to work:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::<AccountId>:user/data-prepper-user"
+      },
+      "Action": "es:ESHttp*",
+      "Resource": [
+        "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/otel-v1*",
+        "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/_template/otel-v1*",
+        "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/_plugins/_ism/policies/raw-span-policy",
+        "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/_alias/otel-v1*",
+        "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/_alias/_bulk"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::<AccountId>:user/data-prepper-user"
+      },
+      "Action": "es:ESHttpGet",
+      "Resource": "arn:aws:es:us-east-1:<AccountId>:domain/<domain-name>/_cluster/settings"
+    }
+  ]
+}
+```
+
+For instructions to configure the domain access policy, see [Resource-based policies
+](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/ac.html#ac-types-resource) in the Amazon OpenSearch Service documentation.
+
+### Fine-grained access control
+
+If your OpenSearch Service domain uses [fine-grained access control
+](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/fgac.html), the OpenSearch sink plugin requires some additional configuration.
+
+#### IAM ARN as master user
+
+If you're using an IAM ARN as the master user, include the `aws_sigv4` option in your sink configuration:
+
+```yaml
+...
+sink:
+    opensearch:
+      hosts: ["https://your-fgac-amazon-opensearch-service-endpoint"]
+      aws_sigv4: true
+```
+
+Run `aws configure` using the AWS CLI to set your credentials to the master IAM user. If you don't want to use the master user, you can specify a different IAM role using the `aws_sts_role_arn` option. The plugin will then use this role to sign requests to the domain sink. The ARN that you specify must be included in the [domain access policy]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/sinks/opensearch/#amazon-opensearch-service-domain-security).
```

Comment:
In the first sentence, do we mean something like "Run `aws configure` using the AWS CLI to use the master IAM user credentials"?

Reply:
Modified.